lib/datadog/core/remote/configuration/content.rb

```diff
@@ -36,6 +38,30 @@ def length
             @length ||= @data.size
           end
 
+          # Sets this configuration as successfully applied.
+          def applied
+            @apply_state = ApplyState::ACKNOWLEDGED
+          end
+
+          # Sets this configuration as not successfully applied, with
+          # a message describing the error.
+          def errored(error_message)
+            @apply_state = ApplyState::ERROR
+            @apply_error = error_message
+          end
+
+          module ApplyState
+            # Default state of configurations.
+            # Set until the component consuming the configuration has acknowledged it was applied.
+            UNACKNOWLEDGED = 1
+
+            # Set when the configuration has been successfully applied.
+            ACKNOWLEDGED = 2
+
+            # Set when the configuration has been unsuccessfully applied.
+            ERROR = 3
```

Comment:
Minor: I'm curious, do we need these integers in the protocol? Could they be symbols (:error, :unacknowledged, ...) instead?

Reply:
They are numbers, as part of the protocol. Backend expects these specific values.